Makefile_defs

```diff
@@ -0,0 +1,31 @@
+# This Makefile defines a bunch of stuff used by the lower-level Makefiles
+# all file paths are from their perspective!
+
+CC = gcc
+
+BIN = ../bin
+BUILD = ../build
+OBJ = $(BUILD)/obj
+
+# this is the name of the lower level directory (e.g. THIS_DIR in the dev_handler folder is "dev_handler")
+THIS_DIR = $(strip $(shell pwd | xargs basename -z))
+
+COMPONENT_DIRS = dev_handler net_handler executor network_switch shm_wrapper runtime_util logger # list of runtime component directories
+OBJ_SUBDIR = $(foreach dir,$(COMPONENT_DIRS),$(OBJ)/$(dir)) # list of subfolders in $(OBJ) for runtime object files
```

Comment:
small nit: suggest putting the in-line comments above the line of code for easier readability

Reply:
agreed, changed!